test/utils.cpp

```diff
@@ -608,32 +591,80 @@ TensorView* biasEpilogue(TensorView* tensor, TensorView* bias) {
       bias->dtype(),
       ", tensor: ",
       tensor->dtype());
-  const auto concrete = TensorDomain::noReductions(
-      TensorDomain::noBroadcasts(tensor->getLeafDomain()));
-  TORCH_CHECK(
-      concrete.size() == 2,
-      "Only tensors with two concrete domains have support for bias epilogue enabled, got ",
-      concrete.size());
   TORCH_CHECK(
       tensor->nDims() >= 2,
       "Tensors to have bias applied needs to have 2 or more domains, got ",
       tensor->nDims());
-  TORCH_CHECK(
-      bias->nDims() == 1,
-      "bias vector must have one dimension, got",
-      bias->nDims());
+
+  const auto concrete = TensorDomain::noReductions(
+      TensorDomain::noBroadcasts(tensor->getLeafDomain()));
 
   TensorView *biasb = nullptr, *biased = nullptr;
-  biasb = broadcast(bias, {false, true});
+
+  switch (concrete.size()) {
+    case 2:
+      // regular matmul (non-strided batch gemm)
+      TORCH_CHECK(
+          bias->nDims() == 1,
+          "bias vector must have one domain, got",
+          bias->nDims());
+      biasb = broadcast(bias, {false, true});
+      break;
+    case 3:
+      // strided batch gemm case
+      TORCH_CHECK(
+          (bias->nDims() == 2 || bias->nDims() == 1),
+          "bias vector must have one (single bias for batch) "
+          "or two (bias for each batch entries)), got",
+          bias->nDims());
+      biasb = broadcast(bias, {false, false, true});
```

Comment:
This only works with `bias->nDims() == 2`, should we handle the two cases separately?

Reply:
You are right - I updated this code and it will be available in the follow up commit. I also added a test case for stirded batch gemm which has a single bias vector that is broadcasted to all batch elements - to be sure that the functionality is handled correctly.